### DISCUSSION TITLE
How to force a query to go to failed state after certain timeout

### DISCUSSION THREAD

**Discussion by pogaku9** [reduxjs/redux-toolkit#2291] [Q&A]
Some of my queries are randomly stuck in `pending` state for a long time, usually our API gateway responds with success or failure within 30sec.

I'd like to know if there is a way to force a `pending` query to `fulfilled` or `failed/error` state after 31sec.

Thanks in advance !

**Comment by phryneas** [comment]
That will be added in #2143, meanwhile you can see some suggestions on doing it in #2079
